src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java

```diff
@@ -0,0 +1,561 @@
+                return 0;
+            }
+
+            if (this.putBatch.size() > 0) {
+                tikv().batchPutAtomic(this.putBatch);
+                this.putBatch.clear();
+            }
+
+            if (this.deleteBatch.size() > 0) {
+                tikv().batchDeleteAtomic(this.deleteBatch);
+                this.deleteBatch.clear();
+            }
+
+            if (this.deletePrefixBatch.size() > 0) {
+                for (ByteString key : this.deletePrefixBatch) {
+                    tikv().deletePrefix(key);
+                }
+            }
+
+            if (this.deleteRangeBatch.size() > 0) {
+                for (Map.Entry<ByteString, ByteString> entry :
+                     this.deleteRangeBatch.entrySet()) {
+                    tikv().deleteRange(entry.getKey(), entry.getValue());
+                }
+            }
+
+            return count;
+        }
+
+        /**
+         * Rollback all updates(put/delete) not committed
+         */
+        @Override
+        public void rollback() {
+            this.putBatch.clear();
+            this.deleteBatch.clear();
+            this.deletePrefixBatch.clear();
+            this.deleteRangeBatch.clear();
+        }
+
+        @Override
+        public Pair<byte[], byte[]> keyRange(String table) {
+            // TODO: get first key and lastkey of fake tikv table
+            return null;
+        }
+
+        /**
+         * Add a KV record to a table
+         */
+        @Override
+        public void put(String table, byte[] key, byte[] value) {
+            this.putBatch.put(this.key(table, key), ByteString.copyFrom(value));
+        }
+
+        @Override
+        public synchronized void increase(String table, byte[] key, byte[] value) {
+            long old = 0L;
+            byte[] oldValue = this.get(table, key);
+            if (oldValue.length != 0) {
+                old = this.l(oldValue);
+            }
+            ByteString newValue = ByteString.copyFrom(
+                                  this.b(old + this.l(value)));
+            tikv().put(this.key(table, key), newValue);
+        }
+
+        @Override
+        public void delete(String table, byte[] key) {
+            this.deleteBatch.add(this.key(table, key));
+        }
+
+        @Override
+        public void deletePrefix(String table, byte[] key) {
+            ByteString deleteKey = this.key(table, key);
+            this.deletePrefixBatch.add(deleteKey);
+        }
+
+        /**
+         * Delete a range of keys from a table
+         */
+        @Override
+        public void deleteRange(String table, byte[] keyFrom, byte[] keyTo) {
+            ByteString startKey = this.key(table, keyFrom);
+            ByteString endKey = this.key(table, keyTo);
+            this.deleteRangeBatch.put(startKey, endKey);
+        }
+
+        @Override
+        public byte[] get(String table, byte[] key) {
+            return tikv().get(this.key(table, key)).toByteArray();
```

Comment:
should we merge cache data (putBatch, delBatch...)?

Reply:
no need. query will handle none committed data in upper layer.

Reply:
is there 'assert !this.hasChanges()' is missed?